### ISSUE TITLE
Front matter missing in several new docs

### ISSUE THREAD

**Issue by afrittoli** [tektoncd/pipeline#5791]
# Expected Behavior

All pipeline documentation pages are shown in the navigation once rendered on tekton.dev/docs

# Actual Behavior

Many markdown files have no front matter specified, which results in links with no text:

<img width="912" alt="image" src="https://user-images.githubusercontent.com/2205608/203843259-91a392b4-7d0a-4e06-a7a1-84fece7fb18b.png">

Note the link to the bundle resolver when I hover with the mouse over the text-less links

# Steps to Reproduce the Problem

1.  Bundle resolver docs: https://raw.githubusercontent.com/tektoncd/pipeline/main/docs/bundle-resolver.md
2.  Task docs: https://raw.githubusercontent.com/tektoncd/pipeline/main/docs/tasks.md

Notice the front-matter in (2), which is missing in (1).

# Additional Info

- Tekton Pipeline version: v0.41.0

/kind documentation

**Comment by afrittoli** [comment]
/cc @abayer 

Once this is fixed on main, we need to cherry-pick to the relevant branches to fix the website

**Comment by afrittoli** [comment]
Complete list of path on the website with no link title:

```
/docs/pipelines/bundle-resolver/
/docs/pipelines/cluster-resolver/
/docs/pipelines/git-resolver/
/docs/pipelines/how-to-write-a-resolver/
/docs/pipelines/hub-resolver/
/docs/pipelines/pipelineresources/
/docs/pipelines/resolution-getting-started/
/docs/pipelines/resolution/
/docs/pipelines/resolver-reference/
/docs/pipelines/trusted-resources/
/docs/pipelines/windows/
```

**Comment by afrittoli** [comment]
https://docs.google.com/spreadsheets/d/1HknnR5FDSPgLJlI7kbtPsUlvL1nGvYeIZA7vKti05os/edit?usp=sharing

**Comment by abayer** [comment]
Is there a way to have a further level of nesting in the sidebar? It seems silly to put the 8 resolver-related links all directly under `Tasks and Pipelines`, but I don't have any idea how the site gets generated and therefore don't know how to do that, or if it's even possible. =)

**Comment by afrittoli** [comment]
Yeah, I think we need further nesting, and not only for the resolvers.

I prepared [this table] as a basis for discussing this exactly - once we agree on the nesting we want we can look for a technical solution on how to achieve it.

**Comment by abayer** [comment]
You forgot the link for `[this table]`. =)

**Comment by afrittoli** [comment]
https://docs.google.com/spreadsheets/d/1HknnR5FDSPgLJlI7kbtPsUlvL1nGvYeIZA7vKti05os/edit?usp=sharing

**Comment by geriom** [comment]
> I think we should agree on a structure and publish it in the dev docs of Tekton pipeline as a reference for future docs. @geriom @tektoncd/core-maintainers

Totally agree. This seems like a great topic to discuss in the docs team meeting. Adding it to the agenda.

**Comment by geriom** [comment]
I going to work on a couple of proposals to improve the way the information is display and share them in the pipelines meeting to discuss.

**Comment by geriom** [comment]
Issue fixed in #6117
